src/cmds/qsub_sup.c

```diff
@@ -258,8 +261,21 @@ get_script(FILE *file, char *script, char *prefix)
 		return (4);
 	}
 
-	while ((in = pbs_fgets_extend(&s_in, &s_len, file)) != NULL) {
+	while ((in = pbs_fgets(&s_in, &s_len, file)) != NULL) {
 		if (!exec && ((sopt = pbs_ispbsdir(s_in, prefix)) != NULL)) {
+			/* Check if this is a directive line that should be extended */
+			extend_len = pbs_extendable_line(in);
+			if (extend_len >= 0) {
+				in[extend_len] = '\0'; /* remove the backslash (\) */
+				extend = pbs_fgets_extend(&extend_in, &extend_len, file);
+				if (extend != NULL) {
+					if (pbs_strcat(&s_in, &s_len, extend) == NULL)
+						return (5);
+				}
+				in = s_in;
+				sopt = pbs_ispbsdir(s_in, prefix);
```

Comment:
I don't see that sopt here being used.  The only time it is used here is when it is called with pbs_ispbsdir() up at line 265.  Also, since s_in would have started with the #PBS directive back there, it wouldn't have changed down here.  The only change to s_in is if we strcat() something onto it.

Reply:
`sopt` is used in `do_dir` immediately below this block. Moved these two lines up inside the `if` however.

Reply:
Oh, sorry about questioning this again.  I hadn't seen your response yet.  Thank you for your answer.  I see why you need to call it twice.